### PR TITLE
[DC-1594] Update GCS retraction to use client library

### PR DESCRIPTION
**Pull request by nishanthpp93**
* [DC-1594] Add GCS helpers in utils
* [DC-1594] Update GCS script to use client library, f-strings

[DC-1594]: https://precisionmedicineinitiative.atlassian.net/browse/DC-1594
[DC-1594]: https://precisionmedicineinitiative.atlassian.net/browse/DC-1594

### CODE REVIEW

File: data_steward/utils/gcs.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    :param bucket: GCS bucket name as string
+    :param prefix: folder prefix for the file to read
+    :param file_name: name of the file to read, with extension
+    :return: list of lines from the file
+    """
+    return retrieve_file_contents(client, bucket, prefix,
+                                  file_name).split(b'\n')
+
+
+def upload_file_to_gcs(client,
```
Comment (dcarbone) [inline data_steward/utils/gcs.py:68]: The implementor should simply do the following:

```python
# should already be defined
bucket = client.get_bucket(bucket_or_name=bucket_name)

# create new blob
blob = bucket.blob(blob_name='/'.join([prefix, file_name]))

# if they already have an io obj
blob.upload_from_file(file_obj=file_obj)

# or, alternatively, have the client do it all for you:
blob.upload_from_filename(filename=filename)

# or, from string
blob.upload_from_string(data=file_contents)
```

File: tests/unit_tests/data_steward/utils/gcs_test.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+from unittest.mock import patch, MagicMock
+from typing import Callable
+
+# Third party imports
+
+# Project imports
+from utils import gcs
+
+
+class GCSTest(TestCase):
```
Comment (dcarbone) [inline tests/unit_tests/data_steward/utils/gcs_test.py:13]: This test class should not be necessary as the implementors should utilize known-functional capabilities within the storage client directly.

**Review by dcarbone — CHANGES_REQUESTED** [review]
I appreciate the efforts here, but I believe that we must stop creating wrappers like this.  The  upstream client created and maintained by Google natively provides the ability to perform all of these actions without additional wrapping, and it'd be best if those were directly used, I think.

I am happy to have further discussions on this, if you like!

File: data_steward/utils/gcs.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Python import
+from io import BytesIO
+
+# Third party imports
+from google.api_core import page_iterator
+
+
+def list_sub_prefixes(client, bucket, prefix):
```
[the remark this answers is not in the document]
Comment (nishanthpp93) [reply data_steward/utils/gcs.py:8]: Post discussion notes: 
1. Helper is useful when there are an inordinate number of blobs but very few 'sub-directories' for a specific path.
2. Add SO link.

File: data_steward/utils/gcs.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        api_request=client._connection.api_request,
+        path=path,
+        items_key='prefixes',
+        item_to_value=lambda _, x: x,
+        extra_params=extra_params,
+    )
+    return list(pg_iterator)
+
+
+def retrieve_file_contents(client, bucket, prefix, file_name):
```
[the remark this answers is not in the document]
Comment (nishanthpp93) [reply data_steward/utils/gcs.py:35]: Post discussion notes: 
1. Remove helper and use built-in methods directly

File: data_steward/utils/gcs.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    gcs_bucket = client.bucket(bucket)
+    blob = gcs_bucket.blob(prefix + file_name)
+    blob.download_to_file(file_bytes)
+    file_bytes.seek(0)
+    file_str = file_bytes.getvalue()
+    return file_str
+
+
+def retrieve_file_contents_as_list(client, bucket, prefix, file_name):
+    """
```
[the remark this answers is not in the document]
Comment (nishanthpp93) [reply data_steward/utils/gcs.py:55]: Post discussion notes:
1. Remove helper

File: data_steward/utils/gcs.py (hunk 1 above)
Comment (nishanthpp93) [reply data_steward/utils/gcs.py:68]: Post discussion notes:
1. Remove helper and use built-in methods directly

File: tests/unit_tests/data_steward/utils/gcs_test.py (hunk 2 above)
Comment (nishanthpp93) [reply tests/unit_tests/data_steward/utils/gcs_test.py:13]: Post discussion notes:
1. Remove unused tests

**Review by dcarbone — APPROVED** [review]
:+1: lgtm, thanks for implementing the changes!  much less code now :)

**Pull request merged**